File: pkg/codegen/testing/utils/testdata/logical-name-pp/nodejs/logical-name.ts
import * as pulumi from "@pulumi/pulumi";
import * as random from "@pulumi/random";

export = async () => {
    const config = new pulumi.Config();
    const configLexicalName = config.require("cC-Charlie_charlie.😃⁉️");
    const resourceLexicalName = new random.RandomPet("aA-Alpha_alpha.🤯⁉️", {prefix: configLexicalName});
    return {
        "bB-Beta_beta.💜⁉": resourceLexicalName.id,
        "dD-Delta_delta.🔥⁉": resourceLexicalName.id,
    };
}

File: pkg/codegen/testing/utils/testdata/logical-name-pp/python/logical-name.py
import pulumi
import pulumi_random as random

config = pulumi.Config()
config_lexical_name = config.require("cC-Charlie_charlie.😃⁉️")
resource_lexical_name = random.RandomPet("aA-Alpha_alpha.🤯⁉️", prefix=config_lexical_name)
pulumi.export("bB-Beta_beta.💜⁉", resource_lexical_name.id)
pulumi.export("dD-Delta_delta.🔥⁉", resource_lexical_name.id)
